Zeeguu Reader/Zeeguu Reader/ZGTextView.swift

```diff
@@ -48,13 +51,31 @@ class ZGTextView: UITextView {
 	
 	override func canPerformAction(action: Selector, withSender sender: AnyObject?) -> Bool {
 		if action == "translate:" {
-			translate(self)
+			if (willInstantlyTranslate) {
+				translate(self)
+				return false
+			}
+			return true
 		}
 		return false
 	}
 	
 	func translate(sender: AnyObject?) {
+		if (isTranslating) {
+			return
+		}
+		isTranslating = true
 		print("translate called for \(self.selectedText()) with context: \"\(self.selectedTextContext())\"")
+		
+		let range = self.selectedRange
+//		let firstCharAfterRange = NSMakeRange(range.location + range.length, 1)
+		
+		let attributes = self.attributedText.attributesAtIndex(range.location + range.length, effectiveRange: nil)
```

Comment:
for readability it would be nice to extract these three lines into their own method which would be called something like `isTheSelectionAlreadyTranslated`